crates/viewer/re_selection_panel/src/item_heading_with_breadcrumbs.rs

```diff
@@ -0,0 +1,293 @@
+                        });
+
+                        // Then the full name of the main item:
+                        last_part_of_item_heading(ctx, viewport, ui, item);
+                    },
+                );
+            }),
+        );
+}
+
+fn item_bread_crumbs_ui(
+    ctx: &ViewerContext<'_>,
+    viewport: &ViewportBlueprint,
+    ui: &mut egui::Ui,
+    item: &Item,
+) {
+    match item {
+        Item::AppId(_) | Item::DataSource(_) | Item::StoreId(_) => {
+            // TODO(emilk): maybe some of these could have breadcrumbs
+        }
+        Item::InstancePath(instance_path) => {
+            let InstancePath {
+                entity_path,
+                instance,
+            } = instance_path;
+
+            if instance.is_all() {
+                // Entity path. Exclude the last part from the breadcrumbs,
+                // as we will show it in full later on.
+                if let [all_but_last @ .., _] = entity_path.as_slice() {
+                    entity_path_breadcrumbs(ctx, ui, None, &EntityPath::root(), all_but_last, true);
+                }
+            } else {
+                // Instance path.
+                // Show the full entity path, and save the `[instance_nr]` for later.
+                entity_path_breadcrumbs(
+                    ctx,
+                    ui,
+                    None,
+                    &EntityPath::root(),
+                    entity_path.as_slice(),
+                    true,
+                );
+            }
+        }
+        Item::ComponentPath(component_path) => {
+            entity_path_breadcrumbs(
+                ctx,
+                ui,
+                None,
+                &EntityPath::root(),
+                component_path.entity_path.as_slice(),
+                true,
+            );
+        }
+        Item::Container(container_id) => {
+            if let Some(parent) = viewport.parent(&Contents::Container(*container_id)) {
+                viewport_breadcrumbs(ctx, viewport, ui, Contents::Container(parent));
+            }
+        }
+        Item::SpaceView(view_id) => {
+            if let Some(parent) = viewport.parent(&Contents::SpaceView(*view_id)) {
+                viewport_breadcrumbs(ctx, viewport, ui, Contents::Container(parent));
+            }
+        }
+        Item::DataResult(view_id, instance_path) => {
+            viewport_breadcrumbs(ctx, viewport, ui, Contents::SpaceView(*view_id));
+
+            let InstancePath {
+                entity_path,
+                instance,
+            } = instance_path;
+
+            if let Some(view) = viewport.view(view_id) {
+                let common_ancestor = instance_path
+                    .entity_path
+                    .common_ancestor(&view.space_origin);
+
+                let relative = &entity_path.as_slice()[common_ancestor.len()..];
+
+                let is_projection = !entity_path.starts_with(&view.space_origin);
```

Comment:
I still haven't figured out how to properly match the breadcrumbs with the blueprint tree panel for projections, since I don't have a good mental model for how projections are shown in the blueprint tree panel.